src/utils/PeopleWithParents.ts

```diff
@@ -0,0 +1,27 @@
+import { Person } from '../types';
+
+export const peopleWithParents = (peopleFromServer: Person[]): Person[] => {
+  const parentsMap: {
+    mothers: { [key: string]: Person };
+    fathers: { [key: string]: Person };
+  } = {
+    mothers: {},
+    fathers: {},
+  };
+
+  peopleFromServer.forEach(person => {
+    if (person.motherName) {
+      parentsMap.mothers[person.motherName] = person;
+    }
+
+    if (person.fatherName) {
+      parentsMap.fathers[person.fatherName] = person;
+    }
+  });
+
+  return peopleFromServer.map(person => ({
+    ...person,
+    mother: parentsMap.mothers[person.motherName || ''] || null,
+    father: parentsMap.fathers[person.fatherName || ''] || null,
```

Comment:
The use of `person.motherName || ''` and `person.fatherName || ''` is a bit redundant since `parentsMap` keys are strings. Consider directly using `person.motherName` and `person.fatherName` without the fallback to an empty string, as the map will handle undefined keys by returning `undefined`, which is then correctly defaulted to `null`.